### DISCUSSION TITLE
How to implement multiple themes

### DISCUSSION THREAD

**Discussion by kdichev** [mui/pigment-css#246] [Discussions]
I have a website that supports multiple themes. In the current setup with Emotion, I achieve this by dynamically injecting different theme objects into the ThemeProvider, based on the domain (e.g., should-have-blue-theme.com or should-have-red-theme.com).

I’m curious if a similar approach can be implemented with Pigment CSS. So far, it seems that only one theme is supported, with options for light and dark mode. Is there a way to extend this functionality to handle multiple themes (light/dark mode included)?

**Comment by aarongarciah** [comment]
As of today, only one theme is supported (with support for light and dark color schemes). We'll provide a mechanism to support several themes in future versions.

cc @brijeshb42 in case you want to share more context.

**Reply by wilson-webdev** [reply]
Do you have an ETA of when this will be available? Plus any insights into how this may work would be great! I am also in the same position as @kdichev whereby I dynamically theme an app based on a runtime condition.

Just an additional note, these values are not known ahead of time, they are stored in the DB. For example, background colour of red for example1.com, but background colour of blue for example2.com
